jmh-benchmarks/src/main/java/org/apache/kafka/jmh/cache/LRUCacheBenchmark.java

```diff
@@ -35,14 +37,29 @@
  * http://hg.openjdk.java.net/code-tools/jmh/file/tip/jmh-samples/src/main/java/org/openjdk/jmh/samples/
  */
 @State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
 public class LRUCacheBenchmark {
 
+    private static final int DISTINCT_KEYS = 10_000;
+
+    private static final String KEY = "the_key_to_use";
+
+    private static final String VALUE = "the quick brown fox jumped over the lazy dog the olympics are about to start";
+
+    private static final String[] KEYS = new String[DISTINCT_KEYS];
+
+    private static final String[] VALUES = new String[DISTINCT_KEYS];
+
     private LRUCache<String, String> lruCache;
 
-    private final String key = "the_key_to_use";
-    private final String value = "the quick brown fox jumped over the lazy dog the olympics are about to start";
     int counter;
 
+    static {
+        for (int i = 0; i < DISTINCT_KEYS; ++i) {
```

Comment:
Why not put this in the `@Setup` method?

Reply:
@bbejeck my reasoning here was that we only need to populate these arrays once per JVM. If I put this in setup we'll potentially (if the number of trials per fork is > 1) duplicate these `String`s a few times. Just needless noise to overwrite the arrays and GC those duplicate `String`s at the beginning of each set of runs (admittedly a trivial amount of noise but given that we're micro-benchmarking why not imo :) ).

Reply:
@original-brownbear I understand that part, but IMHO it's best if we stick to some basic guidelines for our tests/benchmarks.  

To reduce the GC impact how about we decrease the cache size to 5, and the number of records to insert 25 or so? Or just create an array inline with the keys and declare as a private variable?  Either way, we should be able to do the work in the `setUp` method.

Reply:
@bbejeck I think basic guidelines makes sense here :) The choice of a `static` initializer was just out of principle. I just ran the Gradle `jmh` target and it seems we're just running the `@Setup` once per fork anyways => it has no practical impact whatsoever moving this into the `@Setup` => moved it there :)

Reply:
Sorry for being late on this. Populating static variables from a non-static method is generally not a good practice since it's generally not thread-safe. Given how JMH works, it's fine for those fields to be non-static right? Also, it seems more realistic since the data is isolated per run.

Reply:
@ijuma you're right, the `static` only made sense when I had the `static` initializer as well. Removed the `static` on the two arrays now :)